cardano-api/src/Cardano/Api/Script.hs

```diff
@@ -193,36 +192,64 @@ instance TestEquality SimpleScriptVersion where
     testEquality _              _              = Nothing
 
 
-data PlutusScriptVersion lang
-  -- For now, there are no such versions, but it'd be like this:
-  -- PlutusScriptV1 :: PlutusScriptVersion PlutusScriptV1
+data PlutusScriptVersion lang where
+    PlutusScriptV1 :: PlutusScriptVersion PlutusScriptV1
 
 deriving instance (Eq   (PlutusScriptVersion lang))
 deriving instance (Show (PlutusScriptVersion lang))
 
 instance TestEquality PlutusScriptVersion where
-    testEquality lang = case lang of {}
+    testEquality PlutusScriptV1 PlutusScriptV1 = Just Refl
 
 
 data AnyScriptLanguage where
      AnyScriptLanguage :: ScriptLanguage lang -> AnyScriptLanguage
 
 deriving instance (Show AnyScriptLanguage)
 
+instance FromJSON AnyScriptLanguage where
+    parseJSON = withObject "AnyScriptLanguage" $ \o -> do
+      v <- o .: "scriptLanguage"
+      case v of
+        Aeson.String "Plutus" -> return . AnyScriptLanguage $ PlutusScriptLanguage PlutusScriptV1
+        Aeson.String "Simple" -> do
+          ver <- o .: "version"
+          case ver of
+            Aeson.String "SimpleScriptV1" ->
+              return . AnyScriptLanguage $ SimpleScriptLanguage SimpleScriptV1
+            Aeson.String "SimpleScriptV2" ->
+              return . AnyScriptLanguage $ SimpleScriptLanguage SimpleScriptV2
+            err -> fail $ "Error parsing \"version\". Expected \"SimpleScriptV1\" or \"SimpleScriptV2\" \
+                        \ but got: " <> show err
+
+        err -> fail $ "Error parsing \"scriptLanguage\". Expected \"Plutus\" or \"Simple\" \
+                      \ but got: " <> show err
+
+
+instance Aeson.FromJSONKey AnyScriptLanguage where
+    fromJSONKey = Aeson.FromJSONKeyValue parseJSON
```

Comment:
I don't understand how this is supposed to work. A JSON key must be just a JSON string but this reuses the `FromJSON` above right? But that uses a JSON object not a JSON string. Am I missing something?